sidebar.c

```diff
@@ -903,15 +902,111 @@ static bool imap_is_prefix(const char *folder, const char *mbox, size_t *plen)
   if (mutt_str_strncmp(url_m->path, url_f->path, flen) != 0)
     goto done;
 
-  if (url_m->user && !url_f->user)
-    *plen += mutt_str_strlen(url_m->user) + 1;
-  rc = true;
+  plen = strlen(mbox) - mlen + flen;
```

Comment:
Does this mean that `imaps://foo@server.com/Inbox` and `imaps://server.com/Inbox` will match? Is that what we want?